src/main/kotlin/uk/gov/justice/digital/hmpps/nomisprisonerapi/audit/AuditInterceptor.kt

```diff
@@ -3,13 +3,16 @@ package uk.gov.justice.digital.hmpps.nomisprisonerapi.audit
 import org.aspectj.lang.ProceedingJoinPoint
 import org.aspectj.lang.annotation.Around
 import org.aspectj.lang.annotation.Aspect
+import org.slf4j.LoggerFactory
+import org.springframework.core.Ordered
 import org.springframework.core.annotation.Order
 import org.springframework.stereotype.Component
+import org.springframework.transaction.support.TransactionSynchronizationManager
 import uk.gov.justice.digital.hmpps.nomisprisonerapi.repository.StoredProcedureRepository
 
 @Component
 @Aspect
-@Order(1)
+@Order(Ordered.LOWEST_PRECEDENCE) // = 2147483647, lowest precedence so that it runs after the transaction manager
```

Comment:
Actually I believe the transaction manager also has ordering set to LOWEST_PRECEDENCE so it just a happy chance that it now runs first. I couldnt work out how to change this.

Reply:
I wonder if you could inject the transaction manager into this component which would cause Spring to create the transaction manager first?

Reply:
Ive added a @DependsOn which should have the same effect